File: java-framework/src/metreos/util/Assertion.java
/* $Id: Assertion.java 7054 2005-06-07 17:48:33Z wert $
 *
 * Created by wert on Jan 25, 2005
 *
 * Copyright (c) 2005 Metreos, Inc. All rights reserved.
 */

package metreos.util;

/**
 * Portable assertion checking.
 */
public class Assertion
{
	/**
	 * Checks a value for being true and if not it throws an exception
	 * with the specified description in the message.
	 * 
	 * @param value a boolean value which should be true.
	 * 
	 * @param description a description of value (e.g., "x < 100" ).
	 * 
	 * @throws AssertionException when value is not true.
	 */
	public static void check( boolean value, String description )
		throws AssertionException
	{
		if (!value)
			throw new AssertionException( "assertion failed: "+description );
	}
}

File: csharp-framework/Utilities/Assertion.cs
using System;
using System.Runtime.Serialization;

namespace Metreos.Utilities
{
	/// <summary>
	/// Assertion supports making assertions about program state.
	/// </summary>
	[Serializable]
	public class Assertion: Exception
	{
        /// <summary>For serialization use only</summary>
        public Assertion(SerializationInfo info, StreamingContext context) 
            : base(info, context) {}

		private Assertion( string msg )
			: base( msg )
		{
			// nothing to do
		}

		// ////////////// //
		// STATIC METHODS //
		// ////////////// //

		/// <summary>
		/// Checks that a given condition is true. Throws an Assertion exception
		/// if not with the message "assertion failed: "+condStr.
		/// </summary>
		/// <param name="cond">a boolean expression which should normally be true</param>
		/// <param name="condStr">a description of the above expression</param>
		/// <example>Assertion.Check( i == 17, "i == 17" );</example>
		public static void Check( bool cond, string condStr )
		{
			if (!cond)
				throw new Assertion( FAILED+condStr );
		}

		/// <summary>
		/// Checks that a given condition is true. Throws an Assertion exception
		/// if not with the message "assertion failed: "+String.Format( fmtStr, args ).
		/// </summary>
		/// <param name="cond">a boolean expression which should normally be true</param>
		/// <param name="fmtStr">a description of the above expression with argument
		/// substitutions</param>
		/// <param name="args">the args to use for formatting</param>
		/// <example>Assertion.Check( i == 17, "i == 17: i = {0}", i );</example>
		public static void Check( bool cond, string fmtStr, params object[] args )
		{
			if (!cond)
				throw new Assertion( FAILED+String.Format( fmtStr, args ) );
		}

		private const string FAILED = "assertion failed: ";
	}
}
